ga4gh/exceptions.py

```diff
@@ -190,6 +198,84 @@ def getMessage(self):
             self.callSetId, self.variantSetId)
         return message
 
+
+class DataException(BaseServerException):
+    """
+    Exceptions thrown during the server startup, and processing faulty VCFs
+    """
+    message = "Faulty data found or data file is missing."
+
+
+class EmptyDirException(DataException):
+    message = "Diretory empty, no VCF/BCF file was found"
+
+
+class MalFormedException(DataException):
+    """
+    Exception thrown when expected entry or file was missing,
+    or unexpected (meta)data entry was found.
+    """
+    message = "Faulty entry found in data file."
+
+
+class NotIndexedException(MalFormedException):
+    """
+    VCF/BCF files must be indexed, with the same prefix.
+    """
+    def __init__(self, fileName):
+        self.message = (
+            "Indexing file is missing, File {} must be"
+            " indexed.".format(fileName))
+
+
+class OverlappingVcfException(MalFormedException):
+    """
+    Exception thrown when two VCF files within a VariantSet directory
+    contain records for the same contig.
+    """
+    def __init__(self, fileName, contig):
+        self.message = (
+            "VCF file '{}' contains records for contig '{}'. Other files"
+            " in this VariantSet have records for this contig, and overlapping"
+            " VCFs are not permitted.".format(fileName, contig))
+
+
+class InconsistentMetaDataException(MalFormedException):
+    """
+    Exception thrown when two VCF files within a VariantSet direcotry
+    contain different metadata entries.
+    """
+    def __init__(self, fileName):
+        self.fileName = fileName
+        self.message = "Metadata of {} is not consistent with other files"
```

Comment:
I think we should start again and write this in [whitespace](http://en.wikipedia.org/wiki/Whitespace_%28programming_language%29). It would make all these style choices so much easier!